src/main/java/org/spongepowered/api/registry/RegistryEntry.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.registry;
+
+import org.spongepowered.api.ResourceKey;
+
+/**
+ * Represents an entry within a {@link Registry}.
+ *
+ * @param <V> The type of the registry
+ */
+public interface RegistryEntry<V> {
```

Comment:
I think this is going to be the key change here and we need to be totally clear about it.

Is this a replacement for the `CatalogType` interface? Given that, in theory, objects could be in multiple registries with different keys, this makes total sense because an object can no longer be identified by key alone - it's a combination of key and registry (or two keys if you _really_ want), and with scoping, as the server/worlds cycle, so too will the registry entries.

I think, in this case, this should contain the `RegistryKey`.

Reply:
Negative, CatalogType isn't changing. This is simply a useful Key + Value struct for `Registry`. Could actually be tossed but would make `stream` on `Registry` annoying to use.

Reply:
Okay, got it.